src/client/src/fluvio.rs

```diff
@@ -65,6 +65,12 @@ impl Fluvio {
     /// ```
     pub async fn connect_with_config(config: &FluvioConfig) -> Result<Self, FluvioError> {
         let connector = DomainConnector::try_from(config.tls.clone())?;
+        Self::connect_with_connector(connector, config).await
+    }
+    pub async fn connect_with_connector(
+        connector: DomainConnector,
+        config: &FluvioConfig,
+    ) -> Result<Self, FluvioError> {
```

Comment:
Why are we adding this? And more importantly, what is the usage pattern here? Are we expecting typical users to leverage this or is it more of an implementation detail for the WASM library that users will not directly use?